src/org/etools/j1939_84/controllers/part01/Part01Step26Controller.java

```diff
@@ -300,47 +310,586 @@ protected void run() throws Throwable {
+                       })
+                       .forEach(spn -> {
+                           // 6.1.26.12.b. Fail PG query where any bin value received is greater than FAFFh.
+                           addFailure("6.1.26.12.b - Bin value received is greater than 0xFAFF(h)"
+                                   + module.getModuleName() + " for " + spn);
+                       });
+        }
+
+        // 6.1.26.13 Actions7 for MY2022+ Engines
+        // a. DS request message to ECU that indicated support in DM24 for upon request
+        // SP 12730 (GHG Tracking Lifetime Engine Run Time) for each 100hr GHG tracking
+        // PG Label
+        // 64254 GHG Tracking Active 100 Hour Array Data
+        // 64253 GHG Tracking Stored 100 Hour Array Data
+        var ghgTrackingPackets = requestPackets(module.getSourceAddress(),
+                                                GHG_ACTIVE_100_HR,
+                                                GHG_STORED_100_HR)
+                                                                  .stream()
+                                                                  // 6.1.26.13.b.
+                                                                  // Record
+                                                                  // each value
+                                                                  // for use
+                                                                  // in Part 2.
+                                                                  .peek(this::save)
+                                                                  .collect(Collectors.toList());
+
+        if (!ghgTrackingPackets.isEmpty()) {
+            // 6.1.26.13.c. List data received in a table using lifetime, stored 100 hr,
+            // active 100hr for columns, and categories for rows.
+            getListener().onResult(ghgTrackingModule.formatTrackingTable(Stream.concat(ghgTrackingLifetimePackets.stream(),
+                                                                                       ghgTrackingPackets.stream())
+                                                                               .collect(Collectors.toList())));
+        }
+
+        for (int pg : List.of(GHG_ACTIVE_100_HR, GHG_STORED_100_HR)) {
+            GenericPacket ghgTrackingpacketForPg = haveResponseWithPg(ghgTrackingPackets, pg);
+            if (ghgTrackingpacketForPg == null) {
+                // 6.1.26.14.a. For all MY2024+ engines, Fail each PG query where no response was received.
+                if (getEngineModelYear() >= 2024) {
+                    addFailure("6.1.26.10.a - No response was received from "
+                            + module.getModuleName() + "for PG "
+                            + pg);
+                }
+                // 6.1.26.14.b. For MY2022-23 engines, Warn each PG query, where no response was received
+                if (2022 <= getEngineModelYear() && getEngineModelYear() >= 2023) {
```

Comment:
The logic is wrong here. An engine model year of 2024 will be true here.  The logic should be:

```
if (getEngineModelYear >=2022 && getEngineModelYear <=2023) {
```